src/package/Dependencies.js

```diff
@@ -31,6 +31,46 @@ class Dependencies extends AbstractService {
     return path.resolve(rooPath);
   }
 
+  /**
+   * Implementing package pattern ignore
+   * https://github.com/agutoli/serverless-layers/issues/118
+   */
+  async excludePatternFiles() {
+    let filesToIgnore = [];
+    let filesToExclude = [];
+
+    /**
+     * Patterns allows you to define globs that will be excluded / included from the
+     * resulting artifact. If you wish to exclude files you can use a glob pattern prefixed
+     * with ! such as !exclude-me/**. Serverless will run the glob patterns in order so
+     * you can always re-include previously excluded files and directories.
+     *
+     * Reference: https://www.serverless.com/framework/docs/providers/aws/guide/packaging
+     */
+    for (let pattern of this.plugin.service.package.patterns) {
+      if (pattern.startsWith('!')) {
+        const resolvedFiles = await resolveFile(pattern.substr(1), {
+          cwd: this.layersPackageDir
+        });
+        filesToIgnore = filesToIgnore.concat(resolvedFiles);
```

Comment:
why do you, ignore the exclude one ?

Reply:
This plugin doesn't read the `serverless.yaml` file when doing `npm install ` to create the aws lambda layer. Thats why I have to check what files is being ignored to be able to remove unnecessary files.